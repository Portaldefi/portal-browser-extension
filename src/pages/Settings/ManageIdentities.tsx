import React, { useCallback, useEffect } from 'react';
import {
  Grid, Header,
  List
} from 'semantic-ui-react';
import { useNavigate } from 'react-router-dom';
import ConnectionItem from '@/components/Home/RecentConnections/ConnectionItem'
import { useAppSelector } from '@/hooks';

export default () => {
  const identities = useAppSelector(state => state.key.identity);
  const navigate = useNavigate();


  console.log(identities);

  const handleClickConnection = useCallback(() => {
    navigate('/connection-detail')
  }, []);

  return (
    <Grid className='w-100'>
      <Grid.Row centered className='p-none pb-19'>
        <Header as='h1' className='heading'>Manage Identities</Header>
      </Grid.Row>

      <Grid.Row className='p-none'>
        <List className='w-100' style={{ height: "270px", overflowY: "scroll" }}>
          {
            identities.map((identity, index) =>
              <List.Item onClick={() => { }}>
                <List.Content>
                  <ConnectionItem name={"Identity" + index} comment={identity[0].address} />
                </List.Content>
              </List.Item>
            )
          }
        </List>
      </Grid.Row>
    </Grid>
  );
}