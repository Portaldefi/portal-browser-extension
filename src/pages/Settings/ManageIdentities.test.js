import ManageIdentities from './ManageIdentities';
import { create, act } from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store'

describe('ManageIdentitiesComponent', () => {
  const initialState = {
    menu: {
      items: []
    },
    key: {
      privateKey: '',
      privateExtendedKey: '',
      identity: [],
      selectedIdentityId: 0,
      selectedChainId: 0
    }
  }
  const mockStore = configureStore()
  let store


  it('should work', async () => {
    let tree;
    act(() => {
      store = mockStore(initialState)
      tree = create(
        <Provider store={store}>
          <BrowserRouter>
            <ManageIdentities />
          </BrowserRouter>
        </Provider>
      );
    });

    expect(tree).toMatchSnapshot();
  });
  afterAll(() => jest.resetModules());
});