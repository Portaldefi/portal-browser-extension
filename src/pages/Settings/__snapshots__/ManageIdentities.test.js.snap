// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ManageIdentitiesComponent should work 1`] = `
<div
  className="ui grid w-100"
>
  <div
    className="centered row p-none pb-19"
  >
    <h1
      className="ui header heading"
    >
      Manage Identities
    </h1>
  </div>
  <div
    className="row p-none"
  >
    <div
      className="ui list w-100"
      role="list"
      style={
        {
          "height": "270px",
          "overflowY": "scroll",
        }
      }
    />
  </div>
</div>
`;
